algorithms/src/Kokkos_Random.hpp

```diff
@@ -588,10 +588,10 @@ struct Random_XorShift1024_State<false> {
       : state_(&v(state_idx, 0)), stride_(v.stride_1()) {}
 
   KOKKOS_FUNCTION
-  uint64_t operator[](const int i) const { return state_[i * stride_]; }
+  uint64_t operator[](size_t i) const { return state_[i * stride_]; }
```

Comment:
I would be cautious with changing index types for `KOKKOS_FUNCTION` decorated functions.

My experience with large unsigned types on `Cuda` GPU is *don't do it* (you get much more instructions executed, registers and so on).

Could the following be OK with you ?
```
template <typename T>
uint64_t operator[](const T idx) const { return state_[idx * stride_]; }
```

I guess you get the warning from `clang` because it's running for `SERIAL` that has the range policy size type set to `size_t`, correct?

Reply:
Your suggestion would not help with the warning.